astropy/io/fits/header.py

```diff
@@ -938,13 +953,14 @@ def keys(self):
         instance has the same behavior.
         """
 
-        return self.__iter__()
```

Comment:
Why was this changed? It seems unlikely this will affect performance, does it?

Reply:
About this and `.clear` above, I included a commit to change a few non-obvious lines of code:
https://github.com/astropy/astropy/pull/8428/commits/04f2c36b109125554b6bda4e9558d91a0fb2aef2
I think that these changes help a lot in the understanding of the flow in this complex class. For instance in `__init__` I prefer to duplicate three lines instead of having to search each time where some variable is defined.

Reply:
But it makes it harder to understand the semantics. Searching for definitions is an `IDE` thing while understanding the semantics and invariants is a Developer thing. And fits is complicated enough we shouldn't make it harder.

This is just me thinking. If that makes it easier for you to maintain the module go ahead (you're a lot more active than me so it's your judgement call). :)

Reply:
I don't think that the change makes thing harder ;), if you look at the `items`, `keys` and `values` definition it is also much more consistent and easier to understand now.

Reply:
Okay sorry, maybe this was the worst comment for my argument. 😅  

I don't mind the changes to `keys` and `values` (much).